remixer_storage/src/main/java/com/google/android/libraries/remixer/storage/json/SupportedDataType.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.libraries.remixer.storage.json;
+
+import android.graphics.Color;
+import com.google.android.libraries.remixer.ItemListVariable;
+import com.google.android.libraries.remixer.RangeVariable;
+import com.google.android.libraries.remixer.RemixerItem;
+import com.google.android.libraries.remixer.Trigger;
+import com.google.android.libraries.remixer.Variable;
+import com.google.android.remixer.examples.remixerstorage.R;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+/**
+ * Enum that contains all the supported data types for Remixer serialization.
+ *
+ * Each data type has a type identifier string and a {@link ValueConverter} which does most of the
+ * heavywork for serialization.
+ */
+enum SupportedDataType {
+
+  BOOLEAN("boolean", new ValueConverter<Boolean>("boolean") {
+    @Override
+    public Boolean parseValue(JsonElement element) {
+      return element.getAsBoolean();
+    }
+
+    @Override
+    JsonElement valueToJson(Boolean value) {
+      return new JsonPrimitive(value);
+    }
+
+    @Override
+    StoredVariable<Boolean> fromRemixerItem(RemixerItem item) {
+      if (item instanceof Variable) {
+        Variable var = (Variable) item;
+        if (var.getVariableType() == Boolean.class) {
+          StoredVariable<Boolean> storage = new StoredVariable<>();
+          storage.dataType = BOOLEAN.getType();
+          storage.selectedValue = (Boolean) var.getSelectedValue();
+          return storage;
+        }
+      }
+      return null;
+    }
+
+    @Override
+    public StoredVariable<Boolean> createStoredVariable() {
+      return new StoredVariable<>();
+    }
+  }),
+  COLOR("color", new ValueConverter<Integer>("color") {
+    @Override
+    public Integer parseValue(JsonElement element) {
+      JsonObject object = element.getAsJsonObject();
+      return Color.argb(
```

Comment:
Oh interesting, so you have cross-platform concerns to deal with. Very cool, thanks for explanation.